oss_src/sframe/spark_unity.cpp

```diff
@@ -612,11 +612,17 @@ int tordd_main(std::string & _output_directory, size_t & numPartitions, size_t &
 
   auto sframe_len = sframe_ptr->size();
   const std::vector<std::string>& column_names = sframe_ptr->column_names();
+  
+  // Calculate the partition size for the current partition (partId)
   auto partition_size = sframe_len/numPartitions;
-  if(partId < (sframe_len % numPartitions)) { 
+  auto remainder = sframe_len % numPartitions;
+  size_t row_start = partition_size * partId; 
+  if(partId < remainder) { 
     partition_size +=1;
+    row_start = partition_size * partId; 
+  } else { 
+    row_start = partition_size * partId + remainder;
   }
```

Comment:
This whole logic is really annoying. 
What about something simple like:
```
auto partition_size = sframe_len / numPartitions;
row_start = partition_size * partId;
if (partId != numPartitions - 1) {
    row_end = row_start + partition_size;
} else {
    // last partition
    row_end = sframe_len;
}
```
Yes this has a slight degree of imbalance; but no more than numPartitions off. But the logic is much simpler and easier to show that it is correct.